Source/Dafny/Verifier/Translator.cs

```diff
@@ -6260,7 +6260,37 @@ private class WFOptions {
 
       } else if (expr is StmtExpr) {
         var e = (StmtExpr)expr;
+        // If we're inside an "old" expression, then "etran" will know how to translate
+        // expressions. However, here, we're also having to translate e.S, which is a
+        // Statement. Since statement translation (in particular, CallStmt's) works
```

Comment:
Can a CallStmt occur inside a StmtExpr? If not, why is it mentioned here?

Reply:
Yes, a `StmtExpr` is an `Expression` preceded by one of about 5 different kinds of statements (`assert`, `assume`, `reveal`, lemma call, and `calc`). The lemma call was a problem if `old` was placed around the `StmtExpr`. Furthermore, the hints in a `calc` can contain `CallStmt`s, and these were a problem, too. When thinking through various possibly problematic statements, only `CallStmt` seemed problematical, but the fix here takes take of any statement.